service/tablet_allocator.cc

```diff
@@ -2803,11 +2854,17 @@ class load_balancer {
             dst = candidate.dst;
 
             auto& tmap = tmeta.get_tablet_map(source_tablets.table());
-            // If best candidate is co-located sibling tablets, then convergence is re-checked to avoid oscillations.
             if (can_check_convergence && !check_convergence(src_node_info, target_info, source_tablets)) {
-                lblogger.debug("No more candidates. Load would be inverted.");
-                _stats.for_dc(dc).stop_load_inversion++;
-                break;
+                if (_force_capacity_based_balancing) {
+                    lblogger.debug("No more candidates. Load would be inverted.");
+                    _stats.for_dc(dc).stop_load_inversion++;
+                    break;
+                }
+                // If we are performing size-based balancing, the candidate could have been a very large tablet
+                // which would result in load inversion on the target node. Since not all tablets have the same size
+                // and there could be smaller tablets among the remaining candidates, we will continue evaluating
+                // candidates until we drain them all, or achieve balance.
+                continue;
```

Comment:
hmm, I'm worried that this can increase run-time a lot, because `pick_candidate()` is expensive already and if good candidates are burried under a pile of bad candidates, plan making will be slow.